portfolio/src/main/webapp/script.js

```diff
@@ -15,14 +15,111 @@
 /**
  * Adds a random greeting to the page.
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+ //Create a boolean flag to note when the button for random show and random fact have been 
+ //clicked so they will open on first click and close on the second click
+showHasBeenClicked = false;
+factHasBeenClicked = false;
+
+function recommendRandomShow() {
+    //options for randomiser
+  const shows =
+    ['The Good Place', 'Gossip Girl', 'Community', 'Emily in Paris', 'How I met Your Mother'];
+    
+  // Pick a random tv show.
+  const show = shows[Math.floor(Math.random() * shows.length)];
 
   // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+  const showContainer = document.getElementById('tv-show-container');
+  //check if the button has been clicked already and if yes collapse container
+  if (showHasBeenClicked === true){
+    showContainer.innerText = null;
+    showHasBeenClicked = false;
+  }
+  //if not show container and flag it as clicked  
+  else{
+      showContainer.innerText = show;
+      showHasBeenClicked = true;
+  }
+    
+}
+/**
+ * Adds a random fact to the page.
+ */
+function randomFact() {
+  //options for randomiser
+    const facts = 
+    ['favourite color - Purple', 'prefers dogs over cats', 'favourite ice cream flavor - strawberry', 
+    'favourite singer - Lady Gaga', 'still dreams to become an astronaut', '90% of times remembers her dreams'];
+
+  // Pick a random fact.
+  const fact = facts[Math.floor(Math.random() * facts.length)];
+
+  // Add it to the page.
+  const factContainer = document.getElementById('fact-container');
+  //check if clicked and if so show nothinhg
+    if (factHasBeenClicked === true){
+      factContainer.innerText = null;
+      factHasBeenClicked = false;
+    }
+    //else if not clicked show the container and flag as clicked
+    else{
+      factContainer.innerText = fact;
+      factHasBeenClicked = true;
+  }
+}
+
+//Button for 3 sections about me
```

Comment:
Typically we use /**/ style comments for function comments, like the other functions in this file.
Can you also update the functions below.

Reply:
fixed